### PR TITLE
Chart Library: add bar chart component tests + improve data validation

### PR DESCRIPTION
**Pull request by annacmc**
## Proposed changes:

* This PR adds new edge case tests to the bar chart component
* It also resolves any discovered issues causing tests to fail
* It adds data validation to the bar chart

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Testing instructions:

* Run tests for the charts package:
```
cd projects/js-packages/charts
pnpm test
```

Verify that tests run successfully and pass.

In your Jetpack local env, navigate to the Storybook development environment:

```
pnpm i
cd projects/js-packages/storybook
pnpm run storybook:dev
```

Find `JS Packages/Charts/Bar Chart` in the sidebar then navigate down to `Error States` to check the possible error states

image

### CODE REVIEW

File: projects/js-packages/charts/src/components/bar-chart/bar-chart.tsx [hunk 1]
```diff
@@ -23,21 +23,15 @@ type BarChartTooltipData = {
...
-	gridVisibility = 'x',
-	width,
-	height = 400,
-} ) => {
+const BarChart: FC< BarChartProps > = props => {
 	const theme = useChartTheme();
 	const { tooltipOpen, tooltipLeft, tooltipTop, tooltipData, hideTooltip, showTooltip } =
 		useTooltip< BarChartTooltipData >();
 
+	const handleMouseLeave = useCallback( () => {
```
Comment (kangzj) [inline projects/js-packages/charts/src/components/bar-chart/bar-chart.tsx]: I don't think this actually does anything because whenever `hideTooltip` changes, `handleMouseLeave` gets updated.

File: projects/js-packages/charts/src/components/bar-chart/bar-chart.tsx [hunk 2]
```diff
@@ -58,12 +51,37 @@ const BarChart: FC< BarChartProps > = ( {
 		[ showTooltip ]
 	);
 
-	const handleMouseLeave = useCallback( () => {
-		hideTooltip();
-	}, [ hideTooltip ] );
-
+	const {
```
Comment (kangzj) [inline projects/js-packages/charts/src/components/bar-chart/bar-chart.tsx]: It would look a lot clearer to keep those props destructuring as function param IMHO.

File: projects/js-packages/charts/src/components/bar-chart/bar-chart.tsx [hunk 3]
```diff
@@ -102,7 +120,13 @@ const BarChart: FC< BarChartProps > = ( {
 	} ) );
 
 	return (
-		<div className={ clsx( 'bar-chart', className, styles[ 'bar-chart' ] ) }>
+		<div
+			className={ clsx( styles[ 'bar-chart' ], className ) }
+			data-testid="bar-chart"
+			style={ { width, height } }
```
Comment (kangzj) [inline projects/js-packages/charts/src/components/bar-chart/bar-chart.tsx]: Again is this required?

File: projects/js-packages/charts/src/components/bar-chart/bar-chart.tsx (hunk 1 above)
Comment (annacmc) [reply projects/js-packages/charts/src/components/bar-chart/bar-chart.tsx]: Oh yeah you're totally right about that, thanks!

File: projects/js-packages/charts/src/components/bar-chart/bar-chart.tsx [hunk 4]
```diff
@@ -102,7 +118,12 @@ const BarChart: FC< BarChartProps > = ( {
 	} ) );
 
 	return (
-		<div className={ clsx( 'bar-chart', className, styles[ 'bar-chart' ] ) }>
```
Comment (grzegorz-cp) [inline projects/js-packages/charts/src/components/bar-chart/bar-chart.tsx:105]: We added a string class to make the debugging easier for implementers and allow external CSS changes if needed.

Comment (annacmc) [reply projects/js-packages/charts/src/components/bar-chart/bar-chart.tsx:105]: Ah, thank you! 🙏

File: projects/js-packages/charts/src/components/bar-chart/bar-chart.tsx [hunk 5]
```diff
@@ -38,6 +38,10 @@ const BarChart: FC< BarChartProps > = ( {
 	const { tooltipOpen, tooltipLeft, tooltipTop, tooltipData, hideTooltip, showTooltip } =
 		useTooltip< BarChartTooltipData >();
 
+	const handleMouseLeave = () => {
```
Comment (kangzj) [inline projects/js-packages/charts/src/components/bar-chart/bar-chart.tsx]: The wrapper doesn't feel to do anything, so I removed it.

**Review by kangzj — APPROVED** [review]
LGTM! Did a minor change to remove unnecessary code, hope you don't mind 😄 

🚢
